source/guides/azure/infiniband.md

```diff
@@ -0,0 +1,493 @@
+
+Legend:
+
+  X    = Self
+  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
+  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
+  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
+  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
+  PIX  = Connection traversing at most a single PCIe bridge
+  NV#  = Connection traversing a bonded set of # NVLinks
+
+```
+
+## Install UCX-Py and tools
+
+```shell
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+bash Miniconda3-latest-Linux-x86_64.sh
+```
+
+Accept the default and allow conda init to run. Then start a new shell.
+
+Create a conda environment (see ucx-py docs)
+
+```shell
+conda create -n ucxpy -c conda-forge -c rapidsai python=3.7 ipython ucx-proc=*=gpu ucx ucx-py dask distributed numpy cupy pytest pynvml rmm cudf dask-cudf dask-cuda -y
+...
+conda activate ucxpy
+```
+
+Clone ucx-py repo locally
+
+```shell
+git clone https://github.com/rapidsai/ucx-py.git
+cd ucx-py/benchmarks
+```
+
+## Run Tests
+
+Start by running the UCX-Py test suite, from within the `ucx-py` repo:
+
+```shell
+pytest -vs tests/
+pytest -vs ucp/_libs/tests/
+```
+
+Now check to see if IB works, for that you can run some of the benchmarks that we include in UCX-Py, for example:
+
+```shell
+# Let UCX pick the best transport (expecting NVLink when available,
+# otherwise InfiniBand, or TCP in worst case) on devices 0 and 1
+python -m ucp.benchmarks.send_recv --server-dev 0 --client-dev 1 -o rmm --reuse-alloc -n 128MiB
+
+# Force TCP-only on devices 0 and 1
+UCX_TLS=tcp,cuda_copy python -m ucp.benchmarks.send_recv --server-dev 0 --client-dev 1 -o rmm --reuse-alloc -n 128MiB
+```
+
+We expect the first case above to have much higher bandwidth than the second. If you happen to have both
+NVLink and IB connectivity, then you may limit to the specific transport by specifying `UCX_TLS`, e.g.:
+
+```shell
+# NVLink (if available) or TCP
+UCX_TLS=tcp,cuda_copy,cuda_ipc
+
+# InfiniBand (if available) or TCP
+UCX_TLS=tcp,cuda_copy,rc
+
+```
+
+## Run Benchmarks
+
+Finally, let's run the [merge benchmark](https://github.com/rapidsai/dask-cuda/blob/branch-22.12/dask_cuda/benchmarks/local_cudf_merge.py) from `dask-cuda`. This benchmark uses Dask to do a distributed merge across all of the GPUs avaliable
+on your VM. Merge operations in this setting are harder to do because they require communication intensive shuffle
+operations (see the [Dask documentation](https://docs.dask.org/en/stable/dataframe-best-practices.html#avoid-full-data-shuffling)
+for more on this type of operation). In this case, the benchmark will perform an "all-to-all" shuffle, where each dataframe
+will be shuffled before a partition-wise merge. This will require a lot of communication among the GPUs, so network performance
+will be very important.
```

Comment:
> This benchmark uses Dask to perform a merge of two dataframes that are distributed across all the available GPUs on your VM. Merges are a challenging benchmark in a distributed setting since they require communication-intensive shuffle operations of the participating dataframes (see the [Dask documentation](https://docs.dask.org/en/stable/dataframe-best-practices.html#avoid-full-data-shuffling) for more on this type of operation). To perform the merge, each dataframe is shuffled such that rows with the same join key appear on the same GPU. This results in an [all-to-all](https://en.wikipedia.org/wiki/All-to-all_(parallel_pattern)) communication pattern which requires a lot of communication between the GPUs. As a result, network performance will be very important for the throughput of the benchmark.

WDYT?